src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -991,6 +991,29 @@
         (autosign-certificate-request! subject csr settings)
         (fs/delete (path-to-cert-request csrdir subject))))))
 
+(schema/defn ^:always-validate process-csr-deletion!
+  :- {:outcome (schema/enum :success :not-found :error) :message schema/Str}
```

Comment:
As is, this is certainly out-of-place with our existing code, as far as I'm aware - I've observed we have a strong tendency to introduce a `def` for this schema instead of putting it inline like this.

That said, I don't have actually have a problem with it like this, and we don't have any well-defined best practice that I'm aware of.